docs/_docs/user-docs/recipe-headers.md

```diff
@@ -337,6 +337,16 @@ KEEP group:0020
 KEEP group:0028
 ```
 
+**vr**
```

Comment:
Interesting - so @jstorrs I'm wondering, how about for direct attributes doing a more generic select match? E.g., instead of hard coding each attribute (there could be many) to match the overall design of the library, why not have it under a `select` namespace? For example, for vr:

```
BLANK select:vr:TM
```
And then the code is flexible to accept and compare any sets of direct (or even nested) attributes, instead of needing to hard code as we do now. E.g.,

```python

    def select_matches(self, expression):
        """
        Determine whether the element has a specific selected attribute
        """
        attribute, value = expression.split(':', 1)    
        if hasattr(self, attribute.upper(), False):
            return getattr(self, attribute.upper()) == value
        return False
```
And if we wanted to handle nesting, we'd do it based on levels of `:` (e.g., spit into a list and keep going until one is left, and that is the value, and that assumes the value does not have a `:` itself. It might make sense to have a nice clean little class to handle holding the raw value and then parsing it and retrieving the final value. We would instantiate it when we find something that has the "select" prefix.

Does that make sense? Let me know what you think. If we want to include group under this schema we could have an additional lookup of the nesting to match to an identifier the user provides. 

The high level idea is that if we can do something flexible to any attribute someone might be interested in we can avoid a hard coded function for each one.

Reply:
I'll need to think a bit about this (sorry I can be slow). The sort of basic features I would expect to access are what would be seen on the output of say DCMTK's dcmdump. Those will be the group/element numbers, the VR and the VM. But I've personally never encountered needing to access VM in an anonymizer script. Group and VR and pretty useful on their own. VR is nice for tying into lookup tables and databases. Group is helpful to abtract out some modality features that are almost never a PHI concern. So VR is actually the last one I have in mind.

But beyond those my preference actually would be to allow user-provided functions to be registered. They would just look at the pydicom element and return true/false if they wanted to apply. All sorts of complex logic could be put in these sorts of functions. But sequences need some thought. You'd probably also want user-functions to be able to suggest whether they want to descend into sequences or not.

Reply:
okay sounds good - ping me when you want to chat more, or if you want to explicitly layout what you want (and I can try for a first shot).

And no need to apologize - there is lots to do and I totally get it!

Reply:
Ok, I think I see what you are suggesting now and it seems reasonable to me. The rules certainly are more self-explanatory with the select: namespace. I can at least migrate to the namespaces such as select:group:XXXX and select:vr:XX while we think about implementation.

Reply:
And if we think the `<attribute>:<value>` is more elegant, we could have similar, without the select group, and if a `:` (or maybe double) is detected it would be flagged as a select to check. Just an idea! I really like how elegant your original `group:<value>` and `vr:<value>` are and don't want to lose sight of that.

Reply:
I think separating out the select into its own function makes the code more compact and easy to follow.

Reply:
Agree! And then the next question is what the user interaction (what they write in the file) looks like to get there.

Reply:
I think writing "select:property:value" in the file is easy to remember and to follow. Once people know the select:property:value pattern they'll at least know that "group" and "VR" are properties.

Potentially the value component might be extended to handle multiple values in the future (I may have uses for that, but I'll see what I think when I get to that step). Things like ranges "select:group:6000-60FF" to select overlay planes seem likely. IIRC CTP's approach was a special rule to remove overlays but targeting them could be interesting. Or maybe sets to select multiple VR with something like "select:VR:DT|TM|DA").

Maybe this could might even be extended to select:property1:value1:property2:value2. But I'm not sure I have an immediate use for that and I don't want to add things just for the sake of adding things. I'll keep it in the back of my mind in case something comes up where I'm wishing I could apply it.

Reply:
I think what might be a good idea is to step away for a few days, and in the time write down the explicit examples that you really want to work - the use cases that are strongest for you. E.g., likely they are the first set of things, and the expanded one might be developing for something that isn't needed.